Training/Training/Training.c

```diff
@@ -72,122 +63,112 @@ int NumberReverser (int number) {
+      (strOutputs[i] == strExpected[i]) ? printf (ANSI_COLOR_GREEN"\nPassed\n"ANSI_COLOR_RESET) : printf (ANSI_COLOR_RED"\nFailed\n"ANSI_COLOR_RESET);
    }
    do {
       numOutputs = (int*)malloc (numLength * sizeof (int));
    } while (numOutputs == NULL);
    for (int i = 0; i < numLength; i++) {
       printf ("\nInput: %d", numInputs[i]);
       numOutputs[i] = NumberReverser (numInputs[i]);
-      printf ("\nExpected: %d", numExpected[i]);
-      printf ("\nOutput: %d", numOutputs[i]);
-      if (numOutputs[i] == numExpected[i])
-         printf (ANSI_COLOR_GREEN"\nPassed\n"ANSI_COLOR_RESET);
-      else
-         printf (ANSI_COLOR_RED"\nFailed\n"ANSI_COLOR_RESET);
+      printf ("\nExpected: %d\nOutput: %d", numExpected[i], numOutputs[i]);
+      (numOutputs[i] == numExpected[i]) ? printf (ANSI_COLOR_GREEN"\nPassed\n"ANSI_COLOR_RESET) : printf (ANSI_COLOR_RED"\nFailed\n"ANSI_COLOR_RESET);
    }
 }
 
 int main () {
    char ch = '\n', choice[3], * result, term;
    while (1) {
-      printf ("Menu");
-      printf ("\n~~~~");
-      printf ("\n1. Palindrome Checker");
-      printf ("\n2. Number Reverser");
-      printf ("\n3. Run Test Cases");
-      printf ("\n\nEnter your choice: ");
-      fgets (choice, 3, stdin);
+      printf ("Menu\n~~~~\n1. Palindrome Checker\n2. Number Reverser\n3. Run Test Cases\n\nEnter your choice: ");
+      fgets (choice, sizeof (choice), stdin);
       result = strchr (choice, ch);
       if (result == NULL) {
          printf ("Invalid Choice!\n\n");
+         // Draining the unassigned inputs from stdin
          for (; ; ) {
             term = fgetc (stdin);
             if (term == '\n')
                break;
          }
-      } else if (choice[0] == '1') {
-         char phrase[MAX];
-         printf ("\nPalindrome Checker");
-         printf ("\n~~~~~~~~~~~~~~~~~~");
-         printf ("\nEnter a string: ");
-         fgets (phrase, MAX, stdin);
-         result = strchr (phrase, ch);
-         if (result == NULL) {
-            printf ("Input limit exceeded!\n");
-            for (; ; ) {
-               term = fgetc (stdin);
-               if (term == '\n')
+      } else {
+         switch (choice[0]) {
+            case '1': {
+                  char phrase[MAX];
+                  printf ("\nPalindrome Checker\n~~~~~~~~~~~~~~~~~~\nEnter a string: ");
+                  fgets (phrase, MAX, stdin);
+                  result = strchr (phrase, ch);
+                  if (result == NULL) {
+                     printf ("Input limit exceeded!\n");
+                     for (; ; ) {
+                        term = fgetc (stdin);
+                        if (term == '\n')
+                           break;
+                     }
+                  } else {
+                     if (strlen (phrase) == MAX - 1)
+                        phrase[MAX - 1] = '\0';
+                     int pChecker = PalindromeChecker (phrase);
+                     if (pChecker == PALINDROME)

```

Comment:
Can use ternary operator here to simplify this if else block. Apply ternary operator wherever possible to simplify your code.